src/main/java/org/commoncrawl/stormcrawler/news/NewsSiteMapParserBolt.java

```diff
@@ -257,6 +288,30 @@ public void execute(Tuple tuple) {
 
         // send outlinks to status stream
         for (Outlink ol : outlinks) {
+            try {
+                if (!this.crossSubmitAllowed && !crossSubmitCheck(ol, url, metadata)) {
+
+                    String errorMessage = String.format("Cross Submit check failed for %s in %s", ol.getTargetURL(), url);
+                    LOG.error(errorMessage);
+                    ol.getMetadata().setValue(Constants.STATUS_ERROR_SOURCE,
+                            "cross submit check");
+                    ol.getMetadata().setValue(Constants.STATUS_ERROR_MESSAGE, errorMessage);
+                    Values v = new Values(ol.getTargetURL(), ol.getMetadata(),
+                        Status.ERROR);
+                    collector.emit(StatusStreamName, tuple, v);
```

Comment:
Good question, whether URLs failing the cross-submit check should be emitted at all:
- it unnecessary fills up the status index (it's about URLs which are not considered to be fetched)
- the error status is not necessarily a permanent one: depending on the configuration, a error page is retried after some time. Then the cross-submit check would be circumvented after a time delay.